From 17f5277c2f5fe738c81c8927f6bf9383a383c48c Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Tue, 17 Aug 2021 18:05:56 -0700
Subject: [PATCH 27/30] Revert "nbd: add the check to prevent overflow in
 __nbd_ioctl()"

This reverts commit fad7cd3310db3099f95dd34312c77740fbc455e5.

Link: https://github.com/ClangBuiltLinux/linux/issues/1438
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/block/nbd.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/block/nbd.c b/drivers/block/nbd.c
index c5e2b4cd697f..fa4417d56200 100644
--- a/drivers/block/nbd.c
+++ b/drivers/block/nbd.c
@@ -1408,7 +1408,6 @@ static int __nbd_ioctl(struct block_device *bdev, struct nbd_device *nbd,
 		       unsigned int cmd, unsigned long arg)
 {
 	struct nbd_config *config = nbd->config;
-	loff_t bytesize;
 
 	switch (cmd) {
 	case NBD_DISCONNECT:
@@ -1423,9 +1422,8 @@ static int __nbd_ioctl(struct block_device *bdev, struct nbd_device *nbd,
 	case NBD_SET_SIZE:
 		return nbd_set_size(nbd, arg, config->blksize);
 	case NBD_SET_SIZE_BLOCKS:
-		if (check_mul_overflow((loff_t)arg, config->blksize, &bytesize))
-			return -EINVAL;
-		return nbd_set_size(nbd, bytesize, config->blksize);
+		return nbd_set_size(nbd, arg * config->blksize,
+				    config->blksize);
 	case NBD_SET_TIMEOUT:
 		nbd_set_cmd_timeout(nbd, arg);
 		return 0;
-- 
2.33.0

